app/forms/register_user_email_form.rb

```diff
@@ -64,9 +68,11 @@ def password_reset_requested?
   attr_accessor :success, :request_id
 
   def build_user_and_email_address_with_email(email:, email_language:)
+    normalized_email = EmailNormalizer.new(email).normalized_email
+
     self.email_address = user.email_addresses.build(
       user: user,
-      email: email,
+      email: normalized_email,
```

Comment:
If I'm following correctly, this would change so that an email associated with a user's account is not necessarily the email they entered, and prevent them from creating distinct accounts using `+` or floating dot? I wouldn't expect that behavior to be changing as part of this ticket.

Reply:
Maybe we can assign the digest of the normalized email as a separate instance variable, and use that for the rate limiter and `blocked_email_address` method?

```rb
@email_address = user.email_addresses.build(user:, email:)
@normalized_email_digest = SuspendedEmail.generate_email_digest(email)
```

(Optionally moving `generate_email_digest` somewhere more appropriate)

Reply:
(Discussed trade-offs and issue with email_taken?)